source/adaptive_load/nighthawk_service_client.cc

```diff
@@ -0,0 +1,45 @@
+#include "adaptive_load/nighthawk_service_client.h"
+
+#include "external/envoy/source/common/common/assert.h"
+
+namespace Nighthawk {
+
+absl::StatusOr<nighthawk::client::ExecutionResponse> PerformNighthawkBenchmark(
```

Comment:
Looking at this again, I wonder if we could unify this somehow with `RemoteProcessImpl::run(OutputCollector& collector)` https://github.com/envoyproxy/nighthawk/blob/53fa6132757c1a4f40868c94ee058c4d4ffa44cb/source/client/remote_process_impl.cc#L23

They seem similar in some regards, but this has a cleaner interface, and behaves more nicely when it receives unexpected intermediate updates. Maybe it is viable to implement one in terms of the other? (probably out of scope here, but I thought it was worth calling out the similarity.

Reply:
We should definitely unify them. Currently they have slightly different inputs, but we can probably find a common subset.

Reply:
The `OutputCollector` has a method which I think would accept `nighthawk::client::ExecutionResponse::Output`. Hopefully a small tweak would suffice to make this happen. Filed #496 to track, resolving this comment here.